protocol/upgrade/protocol_version.go

```diff
@@ -0,0 +1,128 @@
+package upgrade
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/lavanet/lava/utils"
+	protocoltypes "github.com/lavanet/lava/x/protocol/types"
+	terderminttypes "github.com/tendermint/tendermint/abci/types"
+)
+
+type ProtocolVersion struct {
+	ProviderTarget string
+	ProviderMin    string
+	ConsumerTarget string
+	ConsumerMin    string
+}
+
+var LavaProtocolVersion = ProtocolVersion{
+	ProviderTarget: "0.16.0",
+	ProviderMin:    "0.16.0",
+	ConsumerTarget: "0.16.0",
+	ConsumerMin:    "0.16.0",
+}
+
+type UpgradeManager struct {
+	lock sync.RWMutex
+}
+
+// Returning a new provider session manager
+func NewUpdateManager() *UpgradeManager {
+	return &UpgradeManager{}
+}
+
+func (um *UpgradeManager) SetProtocolVersion(newVersion *protocoltypes.Version) {
+	um.lock.Lock()
+	defer um.lock.Unlock()
+	LavaProtocolVersion.ProviderMin = newVersion.ProviderMin
+	LavaProtocolVersion.ProviderTarget = newVersion.ProviderTarget
+	LavaProtocolVersion.ConsumerTarget = newVersion.ConsumerTarget
+	LavaProtocolVersion.ConsumerMin = newVersion.ConsumerMin
+}
+
+func BuildVersionFromParamChangeEvent(event terderminttypes.Event) bool {
+	foundVersionParam := false
+	var versionValue []byte
+
+	for _, attribute := range event.Attributes {
+		key := string(attribute.Key)
+		value := string(attribute.Value)
+
+		if key == "param" {
```

Comment:
you can simplify this like so:

if key == "param" && value != "version" 
   return false

if key == "value" 
   versionValue = attribute.Value

Reply:
thanks for the suggestion. fixed here: 030a24144b1653ed0663e5c43f345b075624d543